tests/test_emissions.py

```diff
@@ -1,314 +1,295 @@
-        )
-
+from emissions.emission import AtmosphericState, Emission
+
+
+def _scalar(value: np.ndarray | float) -> float:
+    """Convert 0-D/length-1 numpy structures into native floats safely."""
+    arr = np.asarray(value)
+    if arr.size != 1:
+        raise AssertionError("Expected scalar-like value")
+    return float(arr.reshape(-1)[0])
+
+
+class DummyPerformanceModel:
+    """Lightweight stand-in for PerformanceModel with deterministic data."""
+
+    def __init__(self, config_overrides=None, edb_overrides=None):
+        base_config = {
+            'Fuel': 'conventional_jetA',
+            'LTO_input_mode': 'EDB',
+            'EI_NOx_method': 'BFFM2',
+            'EI_HC_method': 'BFFM2',
+            'EI_CO_method': 'BFFM2',
+            'EI_PMvol_method': 'fuel_flow',
+            'EI_PMnvol_method': 'scope11',
+            'CO2_calculation': True,
+            'H2O_calculation': True,
+            'SOx_calculation': True,
+            'APU_calculation': True,
+            'GSE_calculation': True,
+            'LC_calculation': True,
+            'climb_descent_usage': True,
+        }
+        if config_overrides:
+            base_config.update(config_overrides)
+        self.config = base_config
+
+        base_edb = {
+            'fuelflow_KGperS': np.array([0.25, 0.5, 0.9, 1.2], dtype=float),
+            'NOX_EI_matrix': np.array([8.0, 12.0, 26.0, 32.0], dtype=float),
+            'HC_EI_matrix': np.array([4.0, 3.0, 2.0, 1.0], dtype=float),
+            'CO_EI_matrix': np.array([20.0, 15.0, 10.0, 5.0], dtype=float),
+            'PMnvolEI_best_ICAOthrust': np.array([0.05, 0.07, 0.09, 0.12], dtype=float),
+            'PMnvolEI_new_ICAOthrust': np.array([0.04, 0.06, 0.08, 0.11], dtype=float),
+            'PMnvolEIN_best_ICAOthrust': np.array(
+                [1.1e13, 1.2e13, 1.3e13, 1.4e13], dtype=float
+            ),
+            'SN_matrix': np.array([6.0, 8.0, 11.0, 13.0], dtype=float),
+            'PR': np.array([22.0, 22.0, 22.0, 22.0], dtype=float),
+            'ENGINE_TYPE': 'TF',
+            'BP_Ratio': np.array([5.0, 5.0, 5.0, 5.0], dtype=float),
+            'nvPM_mass_matrix': np.array([5.0, 5.5, 6.0, 6.5], dtype=float),
+            'nvPM_num_matrix': np.array([2.0e14, 2.1e14, 2.2e14, 2.3e14], dtype=float),
+            'EImass_max': 8.0,
+            'EImass_max_thrust': 0.575,
+            'EInum_max': 2.4e14,
+            'EInum_max_thrust': 0.575,
+            'EImass_max_alt': 0.85,
+        }
+        if edb_overrides:
+            base_edb.update(edb_overrides)
+        self.EDB_data = base_edb
+
+        self.APU_data = {
+            'fuel_kg_per_s': 0.03,
+            'PM10_g_per_kg': 0.4,
+            'NOx_g_per_kg': 0.05,
+            'HC_g_per_kg': 0.02,
+            'CO_g_per_kg': 0.03,
+        }
+        self.model_info = {
+            'General_Information': {'aircraft_class': 'wide', 'n_eng': 2},
+        }
+
+
+class DummyTrajectory:
```

Comment:
I'm hoping that the new trajectory data structures will make it much easier to construct this sort of test data on demand, in a way that guarantees consistency. It's hard to ensure that this sort of thing stays correct as code changes happen elsewhere.